arch/xtensa/src/esp32/esp32_wireless.c

```diff
@@ -288,6 +370,476 @@ void esp32_phy_disable_clock(void)
   leave_critical_section(flags);
 }
 
+#ifdef CONFIG_ESP32_SUPPORT_MULTIPLE_PHY_INIT_DATA
+
+/****************************************************************************
+ * Name: phy_crc_check
+ *
+ * Description:
+ *   Check the checksum value of data
+ *
+ * Input Parameters:
+ *   data     - Data buffer pointer
+ *   length   - Data length
+ *   checksum - Checksum pointer
+ *
+ * Returned Value:
+ *   OK on success; a negated errno on failure
+ *
+ ****************************************************************************/
+
+static int phy_crc_check(uint8_t *data, const uint8_t *checksum,
+                         size_t length)
+{
+  uint32_t crc_data = 0;
+  crc_data = crc32_le(crc_data, data, length);
+  uint32_t crc_size_conversion = htonl(crc_data);
```

Comment:
Better to use `HTONL` in kernel
```suggestion
  uint32_t crc_size_conversion = HTONL(crc_data);
```

Reply:
Done